analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -221,13 +215,29 @@ class GoMod(
         return graph.breakCycles()
     }
 
-    private fun ModuleInfo.toId(): Identifier =
-        Identifier(
-            type = managerName.takeIf { main } ?: "Go",
-            namespace = "",
-            name = path,
-            version = normalizeModuleVersion(version)
-        )
+    private fun ModuleInfo.toId(): Identifier {
+        val moduleInfo = replace ?: this
+
+        return if (moduleInfo.version.isBlank()) {
+            require(moduleInfo.main) {
+                "Found a local module dependency which is not supported."
+            }
+
+            Identifier(
+                type = managerName,
+                namespace = "",
+                name = moduleInfo.path,
+                version = processProjectVcs(File(moduleInfo.dir!!)).revision
```

Comment:
Would it make sense to add a `requireNotNull(moduleInfo.dir) {}` with an appropriate message instead of using `!!`?

In general, I believe the `require...()` should be `check...()` as this is about program state, not user input.

Reply:
Beyond that, I managed to split-out changed to this function into a separate preceding commit.